### PR TITLE
fix test fail on arm64 mac `panic: permission denied`, removing gomonkey

### LINKED ISSUE #1789: make test failed ` permission denied` on macos 11.6.5
**What happened**:
`make test` failed:
```
=== RUN   Test_GroupClustersWithScore
--- FAIL: Test_GroupClustersWithScore (0.00s)
panic: permission denied [recovered]
        panic: permission denied

goroutine 15 [running]:
testing.tRunner.func1.2({0x101f1a760, 0x102b79c68})
        /Users/didi/sdk/go1.17.5/src/testing/testing.go:1209 +0x348
testing.tRunner.func1(0xc0006036c0)
        /Users/didi/sdk/go1.17.5/src/testing/testing.go:1212 +0x2e4
panic({0x101f1a760, 0x102b79c68})
        /Users/didi/sdk/go1.17.5/src/runtime/panic.go:1052 +0x2ac
github.com/agiledragon/gomonkey/v2.modifyBinary(0x101b39a70, {0xc000132798, 0x18, 0x18})
        /Users/didi/OpenSouorce/karmada/vendor/github.com/agiledragon/gomonkey/v2/modify_binary_darwin.go:9 +0x150
github.com/agiledragon/gomonkey/v2.replace(0x101b39a70, 0x102070018)
        /Users/didi/OpenSouorce/karmada/vendor/github.com/agiledragon/gomonkey/v2/patch.go:255 +0x100
github.com/agiledragon/gomonkey/v2.(*Patches).ApplyCore(0xc0000e1770, {0x101ec08c0, 0x102070030, 0x13}, {0x101ec08c0, 0x102070018, 0x13})
        /Users/didi/OpenSouorce/karmada/vendor/github.com/agiledragon/gomonkey/v2/patch.go:217 +0x214
github.com/agiledragon/gomonkey/v2.(*Patches).ApplyFunc(0xc0000e1770, {0x101ec08c0, 0x102070030}, {0x101ec08c0, 0x102070018})
        /Users/didi/OpenSouorce/karmada/vendor/github.com/agiledragon/gomonkey/v2/patch.go:82 +0x318
github.com/agiledragon/gomonkey/v2.ApplyFunc(...)
        /Users/didi/OpenSouorce/karmada/vendor/github.com/agiledragon/gomonkey/v2/patch.go:24
github.com/karmada-io/karmada/pkg/scheduler/core/spreadconstraint.Test_GroupClustersWithScore(0xc0006036c0)
        /Users/didi/OpenSouorce/karmada/pkg/scheduler/core/spreadconstraint/group_clusters_test.go:176 +0x1608
testing.tRunner(0xc0006036c0, 0x102070020)
        /Users/didi/sdk/go1.17.5/src/testing/testing.go:1259 +0x19c
created by testing.(*T).Run
        /Users/didi/sdk/go1.17.5/src/testing/testing.go:1306 +0x5bc
```

**What you expected to happen**:
test successfully.

**How to reproduce it (as minimally and precisely as possible)**:
make test on macos 11.6.5

**Anything else we need to know?**:
Of course, it's bug of gomonkey, https://github.com/agiledragon/gomonkey/issues/77. But it is a bad experience for karmadaer. So shall we remove `gomonkey`?

**Environment**:
- Karmada version: master
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others: `macOS Big Sur  11.6.5`,  `MacBook Pro (13-inch, M1, 2020)`

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Fix go test fail `panic: permission denied` on arm64 mac

**Which issue(s) this PR fixes**:
Fixes #1789

**Special notes for your reviewer**:
In this PR,
1. remove gomonkey
2. merge duplicate code `calAvailableReplicas`

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
cc @huone1 
Could you please help with it?

**Comment by ikaven1024** [comment]
@huone1 Could you please have a look?

**Comment by huone1** [comment]
> @huone1 Could you please have a look?

sorry, I will review it tonight

File: pkg/scheduler/core/spreadconstraint/group_clusters.go [hunk 1]
```diff
@@ -97,8 +101,10 @@ func groupClustersBasedTopology(
 func groupClustersIngoreTopology(
 	clustersScore framework.ClusterScoreList,
 	rbSpec *workv1alpha2.ResourceBindingSpec,
+	calAvailableReplicasFunc func(clusters []*clusterv1alpha1.Cluster, spec *workv1alpha2.ResourceBindingSpec) []workv1alpha2.TargetCluster,
```
Comment (huone1) [inline pkg/scheduler/core/spreadconstraint/group_clusters.go:104]: the input args calAvailableReplicasFunc is noused

Comment (ikaven1024) [reply pkg/scheduler/core/spreadconstraint/group_clusters.go:104]: yes, it is omissive.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**